### ISSUE TITLE
"schema version mismatch detected" error for system.local and system.peers tables with enabled raft

### ISSUE THREAD

**Issue by temichus** [scylladb/scylla#13299]
## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

"schema version mismatch detected"error in 2023.1.0~rc2-20230302.726f8a090337
where consistent_cluster_management: true by default

but [Jenkins job](https://jenkins.scylladb.com/job/enterprise-2023.1/job/SCT_Enterprise_Features/job/ICS/job/ics-longevity-toggle-strategy-large-partitions-24h-test/3/) has **only 1 failed**  'disrupt_toggle_table_ics' nemesis due to "Schema metadata was not refreshed" and **~200 passed** **during 4 days run**, with error:
```
Command: 'cqlsh --no-color   --request-timeout=120 --connect-timeout=60  -e "ALTER TABLE  scylla_bench.test WITH compaction = {\'class\': \'IncrementalCompactionStrategy\', \'bucket_high\': 1.6, \'bucket_low\': 0.62, \'min_sstable_size\': 440, \'min_threshold\': 2, \'max_threshold\': 60, \'sstable_size_in_mb\': 1609};" 10.4.1.90 9042'

Exit code: 2

Stdout:



Stderr:

&lt;stdin&gt;:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
&lt;stdin&gt;:1:Schema metadata was not refreshed. See log for details.
```

corefile download instructions(1.8 GiB)
```
gsutil cp gs://upload.scylladb.com/core.scylla.112.e526371bcb384f4abc9656b41c569d61.5449.1678889420000000/core.scylla.112.e526371bcb384f4abc9656b41c569d61.5449.1678889420000000.gz .
```

## How frequently does it reproduce?

not reproduced

## Installation details

Kernel Version: 5.15.0-1031-aws
Scylla version (or git commit hash): `2023.1.0~rc2-20230302.726f8a090337` with build-id `efcc950cda22875aa488c5295c6cc68f35d9cc6f`

Cluster size: 5 nodes (i3en.3xlarge)

Scylla Nodes used in this run:
- ics-longevity-toggle-strategy-large-db-node-440d8fc6-5 (52.16.28.157 | 10.4.0.37) (shards: 10)
- ics-longevity-toggle-strategy-large-db-node-440d8fc6-4 (3.250.104.253 | 10.4.3.66) (shards: 10)
- ics-longevity-toggle-strategy-large-db-node-440d8fc6-3 (34.242.52.188 | 10.4.1.90) (shards: 10)
- ics-longevity-toggle-strategy-large-db-node-440d8fc6-2 (3.252.198.163 | 10.4.3.232) (shards: 10)
- ics-longevity-toggle-strategy-large-db-node-440d8fc6-1 (34.241.45.170 | 10.4.2.125) (shards: 10)

OS / Image: `ami-0336a5a646b08dc25` (aws: eu-west-1)

Test: `ics-longevity-toggle-strategy-large-partitions-24h-test`
Test id: `440d8fc6-70d9-499c-8a38-388304731097`
Test name: `enterprise-2023.1/SCT_Enterprise_Features/ICS/ics-longevity-toggle-strategy-large-partitions-24h-test`
Test config file(s):

- [longevity-large-partition-200k_pks-4days.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/4ec1600f015410814cdbe6d94ce29af866d6ccda/test-cases/longevity/longevity-large-partition-200k_pks-4days.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 440d8fc6-70d9-499c-8a38-388304731097`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=440d8fc6-70d9-499c-8a38-388304731097)
- Show all stored logs command: `$ hydra investigate show-logs 440d8fc6-70d9-499c-8a38-388304731097`

## Logs:
- **db-cluster-440d8fc6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/db-cluster-440d8fc6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/db-cluster-440d8fc6.tar.gz)
- **output-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/output-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/output-440d8fc6.log.tar.gz)
- **debug-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/debug-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/debug-440d8fc6.log.tar.gz)
- **events-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/events-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/events-440d8fc6.log.tar.gz)
- **sct-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/sct-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/sct-440d8fc6.log.tar.gz)
- **normal-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/normal-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/normal-440d8fc6.log.tar.gz)
- **argus-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/argus-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/argus-440d8fc6.log.tar.gz)
- **raw_events-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/raw_events-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/raw_events-440d8fc6.log.tar.gz)
- **critical-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/critical-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/critical-440d8fc6.log.tar.gz)
- **warning-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/warning-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/warning-440d8fc6.log.tar.gz)
- **summary-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/summary-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/summary-440d8fc6.log.tar.gz)
- **error-440d8fc6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/error-440d8fc6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/error-440d8fc6.log.tar.gz)
- **monitor-set-440d8fc6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/monitor-set-440d8fc6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/monitor-set-440d8fc6.tar.gz)
- **loader-set-440d8fc6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/loader-set-440d8fc6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/440d8fc6-70d9-499c-8a38-388304731097/20230319_214547/loader-set-440d8fc6.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2023.1/job/SCT_Enterprise_Features/job/ICS/job/ics-longevity-toggle-strategy-large-partitions-24h-test/3/)
</details>

**Comment by mykaul** [comment]
Thanks for looking into it. Do we have any way to know (from the logs) that it did complete later? Anything the client could do differently (extend the default(?) timeout to reduce likelihood, based on no. of nodes, etc) ?

**Comment by gleb-cloudius** [comment]
If the system continue working without printing the message it means schema eventually synchronized. A client should not and cannot do anything.

**Comment by mykaul** [comment]
> If the system continue working without printing the message it means schema eventually synchronized. A client should not and cannot do anything.

If it exists with an error code, clearly we need to do something. @fruch - thoughts?

**Comment by tgrabiec** [comment]
> Check for `.*sing schema commit log` in the logs.

I checked, it is using the schema commitlog, so it's not that.

**Comment by tgrabiec** [comment]
What time did this error occur?

```
&lt;stdin&gt;:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
```

Which log file (there's a dozen linked)?

**Comment by fruch** [comment]
> What time did this error occur?
> 
> ```
> &lt;stdin&gt;:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
> ```
> 
> Which log bunch (there's a dozen linked)?

Should find it in sct-440d8fc6.log.tar.gz

**Comment by tgrabiec** [comment]
But that's earlier, there shouldn't be mismatch at the time cqlsh reports it (2 min later):

```
< t:2023-03-15 14:12:25,465 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG > STDERR: <stdin>:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
< t:2023-03-15 14:12:25,467 f:nemesis.py      l:4209 c:sdcm.nemesis         p:ERROR > <stdin>:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
```

**Comment by tgrabiec** [comment]
@avikivity The kernel stack reporter goes crazy:

```
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff946d28be 0xffffffff946d3061 0xffffffff947baa07 0xffffffff947bb27d 0xffffffff947fbbc3
 0xffffffff946cd622 0xffffffff946cd9b2 0xffffffff946d08c1 0xffffffff946d0982 0xffffffff946d09fd 0xffffffff949ad1ec 0xffffffff94a00099
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff93f201b3 0xffffffff93f4259e 0xffffffff946d291e 0xffffffff946d3061 0xffffffff947baa07
 0xffffffff947bb27d 0xffffffff947fbbc3 0xffffffff946cd622 0xffffffff946cd9b2 0xffffffff946d08c1 0xffffffff946d0982 0xffffffff946d09fd 0xffffffff949ad1ec 0xffffffff94a00099
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff94c0006f 0xffffffff93cc57b4 0xffffffff949b06e3 0xffffffff94a00eeb
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff946df6d0 0xffffffff947cbdcf 0xffffffff947da4e7 0xffffffff947dd7be 0xffffffff947a6bdc 0xffffffff947a6e38 0xffffffff947a6ebd 0xffffffff947a741f 0xffffffff946ff2fa 0xffffffff946ff355 0xffffffff946ff57e 0xffffffff946fe033 0xffffffff946fe416 0xffffffff94c000d9 0xffffffff93cc57b4 0xffffffff949b06e3 0xffffffff94a00eeb
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff93fb4e93 0xffffffff93fb4fd9 0xffffffff93fb5043 0xffffffff946cc3a7 0xffffffff946d0962 0xffffffff946d09fd 0xffffffff949ad1ec 0xffffffff94a00099
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: Reactor stalled for 155 ms on shard 6. Backtrace: 0x51aec83 0x51ae070 0x51af450 0x3cb1f 0x1d40c33 0x1c0ad6c 0x1c52c12 0x1cac3df 0x1caa924 0x1cb761b 0x1cbbb51 0x1aff209 0x1aff0fe 0x1e40343 0x1e43806 0x1da791f 0x1d27922 0x1bd229c 0x1a80992 0x1a852ad 0x19570b3 0x18cd8d8 0x126471a 0x51bf564 0x51c07e7 0x51e15d1 0x51918ba 0x8b12c 0x10cbbf
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: Reactor stalled for 263 ms on shard 6. Backtrace: 0x51aec83 0x51ae070 0x51af450 0x3cb1f 0x13cf282 0x1d806b7 0x1d30708 0x1e4353f 0x1d278e5 0x1bd229c 0x1a80992 0x1a852ad 0x19570b3 0x18cd8d8 0x126471a 0x51bf564 0x51c07e7 0x51e15d1 0x51918ba 0x8b12c 0x10cbbf

```

**Comment by tgrabiec** [comment]
There's an abort on node 3 around that time:

```
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: Aborting on shard 0.
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: Backtrace:
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   0x51aece8
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   0x51e1122
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   /opt/scylladb/libreloc/libc.so.6+0x3cb1f
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   /opt/scylladb/libreloc/libc.so.6+0x8ce5b
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   /opt/scylladb/libreloc/libc.so.6+0x3ca75
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   /opt/scylladb/libreloc/libc.so.6+0x267fb
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   0x1cb2f76
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   0x1c6ca85
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   0x1c6b536
2023-03-15T14:10:20+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]:   0x5437d16

```

**Comment by tgrabiec** [comment]
@temichus Please upload the core dump.

**Comment by tgrabiec** [comment]
Yes

**Comment by fruch** [comment]
@temichus 

SCT is uploading it automatically to Google storage, all the details are part of a coredump event, but I'm guessing  @tgrabiec doesn't know where to look for it...

**Comment by gleb-cloudius** [comment]
Out of memory:

```
                #0  0x00007f0fdad92e5c __pthread_kill_implementation (libc.so.6 + 0x8ce5c)
                #1  0x00007f0fdad42a76 raise (libc.so.6 + 0x3ca76)
                #2  0x00007f0fdad2c7fc abort (libc.so.6 + 0x267fc)
                #3  0x0000000001cb2f77 _ZN8logalloc18allocating_section16on_alloc_failureERNS_6regionE (scylla + 0x1ab2f77)
                #4  0x0000000001c6ca86 _ZN7seastar8futurizeIvE5applyIZZN9row_cache9do_updateIZNS3_6updateENS3_16external_updaterERN7replica8memtableEE4$_21EENS_6futureIvEES5_S8_T_ENKUlvE_clEvEUlvE0_JEEESB_OSC_OSt5tupleIJDpT0_EE (scylla + 0x1a6ca86)
                #5  0x0000000001c6b537 _ZN7seastar20noncopyable_functionIFvvEE17direct_vtable_forIZNS_5asyncIZZN9row_cache9do_updateIZNS5_6updateENS5_16external_updaterERN7replica8memtableEE4$_21EENS_6futureIvEES7_SA_T_ENKUlvE_clEvEUlvE0_JEEENS_8futurizeINSt13invoke_resultISE_JDpT0_EE4typeEE4typeENS_17thread_attributesEOSE_DpOSJ_EUlvE_E4callEPKS2_ (scylla + 0x1a6b537)
                #6  0x0000000005437d17 _ZN7seastar14thread_context4mainEv (scylla + 0x5237d17)

```

**Comment by kostja** [comment]
I wrote a document a while ago [Raft based schema and topology: enablement in Scylla drivers]("https://docs.google.com/document/d/13BTHL42vZBOuIwustZ_GFj_lfVcp77MPwm2it3uMGu0/edit#heading=h.wmb6zzun2mva)

I believe there is no way we will deliver the value of consistent schema management to the clients without changing the drivers. 

This issue is a manifestation of that. Of course, it is both a crash, an overload and a client side schema mismatch. But this is why I think we need to get client side schema mismatch out of the equation.

**Comment by kostja** [comment]
As to the issue itself, I believe nothing much can be done about it - unless we want to explore yet another place where we crash with out of memory.

**Comment by mykaul** [comment]
- do we have any idea why we OOM'ed? Isn't the OOM taking place later, after the schema already converge? (so unrelated to this issue)

**Comment by tgrabiec** [comment]
I don't. Please add links to the issue description.

**Comment by DoronArazii** [comment]
@temichus ^^

**Comment by avikivity** [comment]
> @avikivity The kernel stack reporter goes crazy:
> 
> ```
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff946d28be 0xffffffff946d3061 0xffffffff947baa07 0xffffffff947bb27d 0xffffffff947fbbc3
>  0xffffffff946cd622 0xffffffff946cd9b2 0xffffffff946d08c1 0xffffffff946d0982 0xffffffff946d09fd 0xffffffff949ad1ec 0xffffffff94a00099
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff93f201b3 0xffffffff93f4259e 0xffffffff946d291e 0xffffffff946d3061 0xffffffff947baa07
>  0xffffffff947bb27d 0xffffffff947fbbc3 0xffffffff946cd622 0xffffffff946cd9b2 0xffffffff946d08c1 0xffffffff946d0982 0xffffffff946d09fd 0xffffffff949ad1ec 0xffffffff94a00099
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff94c0006f 0xffffffff93cc57b4 0xffffffff949b06e3 0xffffffff94a00eeb
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff946df6d0 0xffffffff947cbdcf 0xffffffff947da4e7 0xffffffff947dd7be 0xffffffff947a6bdc 0xffffffff947a6e38 0xffffffff947a6ebd 0xffffffff947a741f 0xffffffff946ff2fa 0xffffffff946ff355 0xffffffff946ff57e 0xffffffff946fe033 0xffffffff946fe416 0xffffffff94c000d9 0xffffffff93cc57b4 0xffffffff949b06e3 0xffffffff94a00eeb
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack: 0xffffffffffffff80 0xffffffff93fb4e93 0xffffffff93fb4fd9 0xffffffff93fb5043 0xffffffff946cc3a7 0xffffffff946d0962 0xffffffff946d09fd 0xffffffff949ad1ec 0xffffffff94a00099
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: Reactor stalled for 155 ms on shard 6. Backtrace: 0x51aec83 0x51ae070 0x51af450 0x3cb1f 0x1d40c33 0x1c0ad6c 0x1c52c12 0x1cac3df 0x1caa924 0x1cb761b 0x1cbbb51 0x1aff209 0x1aff0fe 0x1e40343 0x1e43806 0x1da791f 0x1d27922 0x1bd229c 0x1a80992 0x1a852ad 0x19570b3 0x18cd8d8 0x126471a 0x51bf564 0x51c07e7 0x51e15d1 0x51918ba 0x8b12c 0x10cbbf
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: kernel callstack:
> 2023-03-15T14:10:04+00:00 ics-longevity-toggle-strategy-large-db-node-440d8fc6-3     !INFO | scylla[5449]: Reactor stalled for 263 ms on shard 6. Backtrace: 0x51aec83 0x51ae070 0x51af450 0x3cb1f 0x13cf282 0x1d806b7 0x1d30708 0x1e4353f 0x1d278e5 0x1bd229c 0x1a80992 0x1a852ad 0x19570b3 0x18cd8d8 0x126471a 0x51bf564 0x51c07e7 0x51e15d1 0x51918ba 0x8b12c 0x10cbbf
> ```

These need to be decoded against kallsyms. Fortunately s-c-t collects kallsyms in db-cluster.tar.

**Comment by avikivity** [comment]
Unfortunately 5.2 doesn't contain scylladb/seastar@d41af8b59237a949620a9e0b79541b934e0a5375, so those traces could be garbage.

**Comment by fgelcer** [comment]
> @fgelcer please share the kallsym decoder. Better, let's auto-decode those stalls.

https://gist.github.com/fgelcer/3201a4ebf6d49c4e1902876ea63025f3

following changes in SCT (to collect every `kallsyms` after each boot), some things must be adapted in the script...
i will try to decode the ones while we see the `schema version mismatch` and post here

**Comment by mykaul** [comment]
@kostja - seems like the schema mismatch is for system.local and system.peers - why does their schema change, at all? How are the clients relevant for those tables?

**Comment by gleb-cloudius** [comment]
> @kostja - seems like the schema mismatch is for system.local and system.peers - why does their schema change, at all? How are the clients relevant for those tables?

system.local and system.peers is where the driver checks for current schema version for each node. The idea itself is completely broken of course.

**Comment by roydahan** [comment]
Please note that the run @yarongilor refers to, is a run with 1000 keyspaces, each with 1 table (a scale test we have for several years and we do not claim it's a regression)

**Comment by fruch** [comment]
For reference cqlsh has this extra check for schema and it's hard coded 10sec, and he would do this check every time there a CQL command that changes the schema

I've raised 
https://github.com/scylladb/scylla-cqlsh/issues/34

I think we shouldn't be doing this check on every query, and user should be able to select if it need this check or not, and what is the amount of timeout for it.

we do have use cases that schema change might take more than 10 sec, and it might be reasonable

**Comment by gleb-cloudius** [comment]
> For reference cqlsh has this extra check for schema and it's hard coded 10sec, and he would do this check every time there a CQL command that changes the schema

It is 5 sec in the code I was looking at.

**Comment by mykaul** [comment]
> > @kostja - seems like the schema mismatch is for system.local and system.peers - why does their schema change, at all? How are the clients relevant for those tables?
> 
> system.local and system.peers is where the driver checks for current schema version for each node. The idea itself is completely broken of course.

- Isn't it an inheritance from Cassandra?
- Doesn't change the fact that I wouldn't expect a schema mismatch for those tables.

**Comment by gleb-cloudius** [comment]
> > > @kostja - seems like the schema mismatch is for system.local and system.peers - why does their schema change, at all? How are the clients relevant for those tables?
> > 
> > 
> > system.local and system.peers is where the driver checks for current schema version for each node. The idea itself is completely broken of course.
> 
>     * Isn't it an inheritance from Cassandra?

This is Cassandra driver after all.

> 
>     * Doesn't change the fact that I wouldn't expect a schema mismatch for those tables.

There is no schema mismatch for those tables, there is schema mismatch in general. Those tables keep schema version for each node (or rather what each node thinks about other nodes).

**Comment by mykaul** [comment]
OK. Now that I've forced myself to re-read the whole bug, this is probably a duplicate of https://github.com/scylladb/scylla-enterprise/issues/2771 ? - @roydahan ?

**Comment by roydahan** [comment]
Probably.
I see that all "Alter table" commands reported here are related to switching compaction strategy to/from ICS.
And in any case, it should have been opened for enterprise repo and not here.

**Comment by kostja** [comment]
Reclassifying based on this comment: https://github.com/scylladb/scylladb/issues/13299#issuecomment-1484026878 
Not sure it's useful to keep this report at all, it's a very confused/indirect manifestation of the problem - and it could be argued that this manifestation could have a different fix.

**Comment by tgrabiec** [comment]
How do we know this is not a regression, like the one we saw in https://github.com/scylladb/scylladb/issues/13911 ?

**Comment by fgelcer** [comment]
@DoronArazii , we have no priority set on this issue

**Comment by aleksbykov** [comment]
Similar issue like @10701 and this one happened with test:
## Issue description
2 nemesis ran in parallel, On nemesis remove node, repair and then adding node , another nemesis running `ALTER TABLE  keyspace1.standard1 WITH tombstone_gc = {\'mode\': \'repair\'};" 10.4.10.66 9042'`

The command failed with exit code2:
```
Command: '/usr/bin/cqlsh --no-color   --request-timeout=120 --connect-timeout=60  -e "ALTER TABLE  keyspace1.standard1 WITH tombstone_gc = {\'mode\': \'repair\'};" 10.4.10.66 9042'
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > Exit code: 2
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > Stdout:
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > Stderr:
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 
< t:2023-11-04 10:08:03,700 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > <stdin>:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
```

But looks like schema was synced lately, when new node was added to cluster:
```
abykov@vdomresearchbox ~/tmp/d460ffdd-f8f1-452d-a062-05611f67041a $ grep "Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502" db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-*/messages.log
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-10/messages.log:2023-11-04T10:07:37.736+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-10     !INFO | scylla[5880]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-12/messages.log:2023-11-04T10:07:37.898+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-12     !INFO | scylla[8944]:  [shard 0:main] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-14/messages.log:2023-11-04T10:16:17.869+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-14     !INFO | scylla[5892]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-14/messages.log:2023-11-04T10:16:17.869+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-14     !INFO | scylla[5892]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-14/messages.log:2023-11-04T10:16:17.869+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-14     !INFO | scylla[5892]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-14/messages.log:2023-11-04T10:16:17.869+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-14     !INFO | scylla[5892]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-14/messages.log:2023-11-04T10:16:17.869+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-14     !INFO | scylla[5892]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-14/messages.log:2023-11-04T10:16:17.869+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-14     !INFO | scylla[5892]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-4/messages.log:2023-11-04T10:07:37.991+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-4     !INFO | scylla[5997]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
db-cluster-d460ffdd/longevity-parallel-topology-schema--db-node-d460ffdd-9/messages.log:2023-11-04T10:07:38.002+00:00 longevity-parallel-topology-schema--db-node-d460ffdd-9     !INFO | scylla[5880]:  [shard 0:strm] schema_tables - Schema version changed to dfd0bd95-6683-3913-a8fc-8da170df8502
```

While same command in 10 minutes for another table finished without errors

## Impact

Only failed cqlsh command.

## How frequently does it reproduce?

1 time durint test run

## Installation details

Kernel Version: 5.15.0-1049-aws
Scylla version (or git commit hash): `5.5.0~dev-20231027.227136ddf54f` with build-id `420b5e5ef0c99c93991034185a1061ec90ab2c43`





Cluster size: 5 nodes (i4i.2xlarge)

Scylla Nodes used in this run:
- longevity-parallel-topology-schema--db-node-d460ffdd-9 (54.154.147.36 | 10.4.10.66) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-8 (34.242.78.226 | 10.4.9.72) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-7 (63.32.100.237 | 10.4.8.64) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-6 (52.211.113.220 | 10.4.9.120) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-5 (3.253.51.245 | 10.4.11.55) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-4 (3.254.107.237 | 10.4.11.66) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-3 (3.249.137.129 | 10.4.8.227) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-2 (34.244.182.172 | 10.4.8.124) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-17 (34.244.91.207 | 10.4.9.5) (shards: -1)
- longevity-parallel-topology-schema--db-node-d460ffdd-16 (3.254.64.158 | 10.4.10.127) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-15 (54.171.253.122 | 10.4.11.246) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-14 (54.194.136.19 | 10.4.10.41) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-13 (18.203.157.148 | 10.4.9.152) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-12 (34.244.49.82 | 10.4.11.101) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-11 (34.245.207.219 | 10.4.9.21) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-10 (3.252.84.192 | 10.4.11.19) (shards: 7)
- longevity-parallel-topology-schema--db-node-d460ffdd-1 (34.247.48.55 | 10.4.10.5) (shards: 7)


OS / Image: `ami-0445669e3c59c66ab` (aws: undefined_region)

Test: `longevity-schema-topology-changes-12h-test`
Test id: `d460ffdd-f8f1-452d-a062-05611f67041a`
Test name: `scylla-master/longevity/longevity-schema-topology-changes-12h-test`
Test config file(s):

- [longevity-parallel-schema-changes-12h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/932d90acc6fe19dec3e1dc5eb58598919b9f4dae/test-cases/longevity/longevity-parallel-schema-changes-12h.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor d460ffdd-f8f1-452d-a062-05611f67041a`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=d460ffdd-f8f1-452d-a062-05611f67041a)
- Show all stored logs command: `$ hydra investigate show-logs d460ffdd-f8f1-452d-a062-05611f67041a`


## Logs:
- **db-cluster-d460ffdd.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/db-cluster-d460ffdd.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/db-cluster-d460ffdd.tar.gz)
- **sct-runner-events-d460ffdd.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/sct-runner-events-d460ffdd.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/sct-runner-events-d460ffdd.tar.gz)
- **sct-d460ffdd.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/sct-d460ffdd.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/sct-d460ffdd.log.tar.gz)
- **loader-set-d460ffdd.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/loader-set-d460ffdd.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/loader-set-d460ffdd.tar.gz)
- **monitor-set-d460ffdd.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/monitor-set-d460ffdd.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d460ffdd-f8f1-452d-a062-05611f67041a/20231105_051527/monitor-set-d460ffdd.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-schema-topology-changes-12h-test/21/)
[Argus](https://argus.scylladb.com/test/024e78ba-3671-4fd0-bafc-b48b2a978f59/runs?additionalRuns[]=d460ffdd-f8f1-452d-a062-05611f67041a)
</details>

**Comment by mykaul** [comment]
@kostja - can we look at the latest failure?

**Comment by kostja** [comment]
@mykaul I checked the logs. I want to see https://github.com/scylladb/scylladb/issues/16004 and https://github.com/scylladb/scylladb/commit/3d4398d1b2b9bc2139718e49a806214171ee081f before I look into this again, and I hope I won't have to once the above two are in.

**Comment by mykaul** [comment]
> @mykaul I checked the logs. I want to see #16004 and [3d4398d](https://github.com/scylladb/scylladb/commit/3d4398d1b2b9bc2139718e49a806214171ee081f) before I look into this again, and I hope I won't have to once the above two are in.

Aren't both in? Notice the latest failure is on master.

**Comment by mykaul** [comment]
@kostja - what's the next step here?

**Comment by kbr-scylla** [comment]
Could be https://github.com/scylladb/scylladb/issues/17493

**Comment by kbr-scylla** [comment]
We need to retest

**Comment by kbr-scylla** [comment]
> Could be https://github.com/scylladb/scylladb/issues/17493


Actually probably not. And if yes then https://github.com/scylladb/scylladb/issues/17493 is already fixed.

After reading previous comments I think it's probably https://github.com/scylladb/scylla-enterprise/issues/4082#issuecomment-2147840566 --- slow schema apply (which driver times out on) due to overloaded cluster and schema changes using wrong SG.

Marking as `blocked` by https://github.com/scylladb/scylla-enterprise/issues/4082
